apps/cli/test/builder/diretory.test.ts

```diff
@@ -0,0 +1,68 @@
+import fs from "fs-extra";
+import path from "path";
+import { afterAll, describe, expect, it } from "vitest";
+import { build } from "../../src/builder/directory";
+import { DirectoryDriveConfig } from "../../src/config";
+
+describe("directory_builder", () => {
+    const image = "cartesi/sdk:0.11.0";
+
+    it("non-existent directory", async ({ task }) => {
+        const destination = path.join(__dirname, "output", task.suite!.name);
+        const drive: DirectoryDriveConfig = {
+            builder: "directory",
+            directory: path.join(__dirname, "data", "invalid"),
+            extraSize: 0,
+            format: "ext2",
+        };
+        await expect(build(task.id, drive, image, destination)).rejects.toThrow(
+            "no such file or directory",
+        );
+    });
+
+    it("empty directory", async ({ task }) => {
+        const destination = path.join(__dirname, "output", task.suite!.name);
+        const drive: DirectoryDriveConfig = {
+            builder: "directory",
+            directory: path.join(__dirname, "data", "empty"),
+            extraSize: 0,
+            format: "ext2",
+        };
+        await expect(build(task.id, drive, image, destination)).rejects.toThrow(
+            "too few blocks",
+        );
+    });
+
+    it("empty directory with extra size", async ({ task }) => {
+        const destination = path.join(__dirname, "output", task.suite!.name);
+        const drive: DirectoryDriveConfig = {
+            builder: "directory",
+            directory: path.join(__dirname, "data", "empty"),
+            extraSize: 1024 * 1024, // 1Mb
+            format: "ext2",
+        };
+        await build(task.id, drive, image, destination);
+        const filename = path.join(destination, `${task.id}.ext2`);
+        const stat = fs.statSync(filename);
+        expect(stat.size).toEqual(1069056);
+    });
+
+    it("success", async ({ task }) => {
+        const destination = path.join(__dirname, "output", task.suite!.name);
+        const drive: DirectoryDriveConfig = {
+            builder: "directory",
+            directory: path.join(__dirname, "data", "sample1"),
+            extraSize: 0,
+            format: "ext2",
+        };
+        await build(task.id, drive, image, destination);
+        const filename = path.join(destination, `${task.id}.ext2`);
+        const stat = fs.statSync(filename);
+        expect(stat.size).toEqual(32768);
```

Comment:
Where this  `32768` come from?

Reply:
from testing. these sizes are kind of unpredictable